app/Http/Controllers/Api/AuthController.php

```diff
@@ -4,17 +4,26 @@
 
 use App\Http\Controllers\Controller;
 use App\Http\Requests\StorePostRequest;
+use App\Services\UserService;
 use Illuminate\Support\Facades\Hash;
 use App\Models\User;
+use function Laravel\Prompts\password;
 
 class AuthController extends Controller
 {
+    protected UserService $userService;
+
+    public function __construct(UserService $userService)
+    {
+        $this->userService = $userService;
+    }
+
     public function register(StorePostRequest $request)
     {
-        $user = User::create([
-            'name' => $request->name,
-            'email' => $request->email,
-            'password' => Hash::make($request->password),
+        $user = $this->userService->store([
```

Comment:
you can do $user = $this->userService->store($request->validated());

$request->all() - returns array with all data from request
$request->validated() - returns array with all validated data from request